boards: arm: add Renesas R-Car M3ULCB board support (#40697)
zephyrproject-rtos/zephyr: boards/arm/rcar_m3ulcb/doc/rcar_m3ulcb.rst

| CLOCK     | clock_control                |                                |
+-----------+------------------------------+--------------------------------+
| GPIO      | gpio                         |                                |
+-----------+------------------------------+--------------------------------+
| UART      | uart                         | serial port-polling            |
+           +                              +                                +
|           | FT232RQ / CP2102             | serial port-interrupt          |
+-----------+------------------------------+--------------------------------+
| CAN       | can                          | normal mode                    |
+           +                              +                                +
|           | TCAN332GDCNT                 | loopback mode                  |
+-----------+------------------------------+--------------------------------+
| I2C       | i2c                          | interrupt driven               |
+-----------+------------------------------+--------------------------------+

It's also currently possible to write on the ram console.

> More features will be supported soon.

Reviewer: ```suggestion
Other hardware features are not currently supported by the port.
```

This is more or less our standard phrase.

Connections and IOs
===================

M3ULCB Board
------------

Here are official IOs figures from eLinux for M3ULCB board:

`M3SK top view`_

`M3SK bottom view`_

Kingfisher Infotainment daughter board
--------------------------------------

When connected to Kingfisher Infotainment board through COMExpress connector, the board is exposing much more IOs.

Here are official IOs figures from eLinux for Kingfisher Infotainment board:

`Kingfisher top view`_

`Kingfisher bottom view`_

GPIO
----

> By running Zephyr on M3ULCB, the software readable push button 'SW3' can be used as input, and the software contollable LED 'LED5' can be used as output.

Reviewer: ```suggestion
By running Zephyr on M3ULCB, the software readable push button 'SW3' can be used as input, and the software controllable LED 'LED5' can be used as output.
```

UART
----

> M3ULCB board is providing two serial ports, only one is commonly available on the board, however, the second one can be made available either by welding components or by plugging the board on a Kingfisher Infotainment daughter board.

Reviewer: ```suggestion
M3ULCB board is providing two serial ports, only one is commonly available on the board, however, the second one can be made available either by soldering in components or by plugging the board on a Kingfisher Infotainment daughter board.
```

Here is information about these serial ports:

+--------------------+-------------------+--------------------+-----------+--------------------------------------+
| Physical Interface | Physical Location | Software Interface | Converter | Further Information                  |
+====================+===================+====================+===========+======================================+
| CN12 DEBUG SERIAL  | ULCB Board        | SCIF2              | FT232RQ   | Used by U-BOOT & Linux               |
+--------------------+-------------------+--------------------+-----------+--------------------------------------+

> | CN10 DEBUG SERIAL  | ULCB Board        | SCIF1              | CP2102    | Non-welded                           |

Reviewer: ```suggestion
| CN10 DEBUG SERIAL  | ULCB Board        | SCIF1              | CP2102    | Non-populated                           |
```
